experimental/apps-mcp/lib/providers/io/state.go

```diff
@@ -0,0 +1,320 @@
+package io
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/databricks/cli/experimental/apps-mcp/lib/fileutil"
+	"github.com/zeebo/blake3"
+)
+
+const StateFileName = ".edda_state"
+
+// StateType represents the lifecycle state of a scaffolded project.
+type StateType string
+
+const (
+	StateScaffolded StateType = "Scaffolded"
+	StateValidated  StateType = "Validated"
+	StateDeployed   StateType = "Deployed"
+)
+
+// String returns the string representation of the state.
+func (s StateType) String() string {
+	return string(s)
+}
+
+// IsValid checks if the state type is a valid value.
+func (s StateType) IsValid() bool {
+	switch s {
+	case StateScaffolded, StateValidated, StateDeployed:
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
Non-blocking: this Edda state machine is interesting, we should look at how it would translate to the CLI world.